### DISCUSSION TITLE
Importing Rapier Physics Lib

### DISCUSSION THREAD

**Discussion by wirmachenbunt** [cables-gl/cables_docs#923] [Answers & Questions] [answered]
i tried using Rapier Physics with cables but failed so far to import and use the lib.
decode.gl made a pretty good tutorial regarding libs and the difference between IIFE and modules. i'm used to modules and rapier seems working this way, so i went this way, trying to copy paste the lib from sdelivr.com, but no luck so far.

https://cdn.jsdelivr.net/npm/@dimforge/rapier3d@0.14.0/rapier.min.js
under this url, there is not the full lib

i don't really understand how to make it work. pretty tricky to use libs in cables since most stuff i found are modules based. any hint would be welcome

**Comment by pandrr** [comment, marked as answer]
yeah, js library stuff is horrible and complicated and a big pain.

you have to build it yourself using a web packager like webpack

i already have rapier.js working in cables. but no time/money to work on it

here is the js, than can be used as op asset lib

[rapierpkg.js.zip](https://github.com/user-attachments/files/17843334/rapierpkg.js.zip)

**Comment by wirmachenbunt** [comment]
<img width="1029" alt="image" src="https://github.com/user-attachments/assets/2b24af88-966d-4141-b383-3e9dadefe379">
this is what i was looking for. amazing.
how did you make this js file from rapier ?
